src/components/app/exporters/googleSheets/SpreadsheetExporter.vue

```diff
@@ -0,0 +1,80 @@
+<template>
+  <v-form
+    ref="vForm"
+    v-model="validated"
+  >
+    <div v-if="isTokenValid">
+      <v-checkbox
+        v-model="exporter.active"
+        class="ma-0"
+        label="Active"
+        @change="changed = true"
+      />
+      <sheets-combobox
+        v-model="exporter.options.spreadsheetId"
+        :credentials="exporter.options.credentials"
+        @change="changed = true"
+      />
+      <v-btn
+        :disabled="!readyToSave.value"
+        color="primary"
+        @click="submit()"
+      >
+        Save
+      </v-btn>
+    </div>
+
+    <div v-else>
+      <v-btn
+        color="primary"
+        @click="login()"
+      >
+        Login to Google
+      </v-btn>
+    </div>
```

Comment:
If the user is not logged in, this button will display.

Both this button and the dropdown are under the `vForm` since the two are changing the `config`. The login button produces the `credentials` and saves them in the `config`, then the dropdown uses the `credentials` to get a list of spreadsheets from the user account.

Later, with *Vue 3* we can use [Suspance](https://v3.vuejs.org/guide/component-dynamic-async.html#using-with-suspense)

Reply:
For now I changed it to be statefull by `enum`, the commit is on the queue.